types.go

```diff
@@ -156,5 +159,13 @@ type Error struct {
 
 // Error returns a textual description of the test error.
 func (err Error) Error() string {
+	errValues := [3]string{err.Body, err.Message, err.Type}
```

Comment:
Could probably have a series of stanzas like this:

```go
if str := strings.TrimSpace(err.XYZ); str != "" {
    return str
}
```

With a fallback of `return strings.TrimSpace(err.XYZ)` at the very end.

Reply:
if we do this then we modify the existing behaviour. In this case what you suggested we are returning the trimmed string, not the original, hence need to modify existing tests as well. My original intention was to use trimming only for checking whether the string is empty or not _but_ return the _original_ string. I can go ahead with your suggestion, just double checking :)

Reply:
Ah, you are correct! I misread it and thought that `str` was being updated.

Reply:
on it 👌